pkg/sql/alter_index_visible.go

```diff
@@ -0,0 +1,49 @@
+// Copyright 2022 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package sql
+
+import (
+	"context"
+
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog"
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog/tabledesc"
+	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
+	"github.com/cockroachdb/cockroach/pkg/util/errorutil/unimplemented"
+)
+
+type alterIndexVisibleNode struct {
+	n         *tree.AlterIndexVisible
+	tableDesc *tabledesc.Mutable
+	index     catalog.Index
+}
+
+func (p *planner) AlterIndexVisible(
+	ctx context.Context, n *tree.AlterIndexVisible,
+) (planNode, error) {
+	return nil, unimplemented.Newf(
+		"Not Visible Index",
+		"altering an index to visible or not visible is not supported yet")
+}
+
+// TODO (wenyihu6): Double check on whether we should have ReadingOwnWrites. I
+// don't think changing visibility performs multiple KV operations on descriptor
+// and expects to see its own writes. But I'm not certain since renameIndexNode
+// is also implementing this interface method.
+// func (n *alterIndexVisibleNode) ReadingOwnWrites() {}
+
+func (n *alterIndexVisibleNode) startExec(params runParams) error {
+	return unimplemented.Newf(
+		"Not Visible Index",
```

Comment:
nit: define the `feature` field value as a constant, and consider replacing it with "INDEX NOT VISIBLE" perhaps? I see that we usually define these as some kind of pseudo-SQL syntax but that's not always true, there's no strict convention for what the string should look like.

Reply:
Followup: in https://github.com/cockroachdb/cockroach/pull/86032 I see you use `ALTER INDEX VISIBILITY`, that's fine too.